src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+    "click .create-polygon": "onClickCreatePolygon",
+    "click .edit-geometry": "onClickEditGeometry",
+    "click .delete-geometry": "onClickDeleteGeometry",
+    "click .colorpicker": "onClickColorpicker",
+    "change input[name='geometry']": "onIconSelection"
+  },
+  initialize: function() {
+    var self = this;
+
+    this.map = this.options.map;
+    this.workingGeometry = null;
+    this.numVertices = 0;
+    this.layerType = null;
+    this.iconUrl = null;
+    this.existingLayerView = null;
+    this.isEditing = false;
+    this.editingLayerGroup = new L.FeatureGroup();
+    this.DRAWING_DEFAULTS = {
+      color: "#f06eaa",
+      opacity: 0.5,
+      fillColor: "#f06eaa",
+      fillOpacity: 0.2,
+      fill: {
+        color: "fillColor",
+        opacity: "fillOpacity"
+      },
+      stroke: {
+        color: "color",
+        opacity: "opacity"
+      }
+    };
+    this.colorpickerSettings = {
+      editMode: "fill",
+      color: this.DRAWING_DEFAULTS.color,
+      opacity: this.DRAWING_DEFAULTS.opacity,
+      fillColor: this.DRAWING_DEFAULTS.fillColor,
+      fillOpacity: this.DRAWING_DEFAULTS.fillOpacity
+    };
+
+    // Init the colorpicker
+    $("body").append("<div class='toolbar-colorpicker-container'></div>");
+    $(".toolbar-colorpicker-container").spectrum({
+      flat: true,
+      showButtons: false,
+      showInput: true,
+      showAlpha: true,
+      
+      // Convert to rgba() format
+      color: tinycolor(self.colorpickerSettings.fillColor)
+        .setAlpha(self.colorpickerSettings.fillOpacity)
+        .toRgbString(),
+      move: function(color) {
+        if (self.placeDetailView) {
+          self.placeDetailView.onModified();
+        }
+        if (self.editingLayerGroup.getLayers().length > 0) {
+          if (self.colorpickerSettings.editMode === "fill") {
+            self.editingLayerGroup.getLayers()[0].setStyle({
+              fillColor: color.toHexString(),
+              fillOpacity: color.getAlpha()
+            });
+            self.colorpickerSettings.fillColor = color.toHexString();
+            self.colorpickerSettings.fillOpacity = color.getAlpha();
+          } else if (self.colorpickerSettings.editMode === "stroke") {
+            self.editingLayerGroup.getLayers()[0].setStyle({
+              color: color.toHexString(),
+              opacity: color.getAlpha()
+            });
+            self.colorpickerSettings.color = color.toHexString();
+            self.colorpickerSettings.opacity = color.getAlpha();
+          }
+        }
+      },
+      change: function() {
+        //console.log("color change");
+      }
+    });
+
+    $(".sp-picker-container")
+      .addClass("hidden")
+      .prepend(Handlebars.templates["colorpicker-controls"]());
+
+    $(".sp-choose").on("click", function(evt) {
+      self.onClickColorEditModeChange(evt);
+    });
+
+    var generateGeometry = function(layer) {
+      var buildCoords = function(layer) {
```

Comment:
Yes, very good point-- looking back at this I have no idea why I set it up this way. Resolved.